### PR TITLE
Fixes for updates in Suite methods for Sensitivity Analysis

### LINKED ISSUE #889: Make package work again after breaking change in R API on Sensitivity analysis
@IndrajeetPatil Do you want to check this out? We could discuss this you and I if you want.

after the renaming of some of our methods to be explicit about parameter (SA parameter name vs ParameterPath), the code in ospsuite-r needs to be updated and potentially new API methods created to help support new features

### LINKED ISSUE #896: `DataSet$setValues` fails when only one data point is set
```
ds1 <- DataSet$new(name = "Sim1")
ds1$setValues(xValues = 0, yValues = 1, yErrorValues = 2.747)

Error in clrCallStatic(clrFacadeTypeName, "SetFieldOrProperty", objOrType, : 
Type: System.ArgumentException
Message: Object of type 'System.Double' cannot be converted to type 'System.Double[]'.
Method: System.Object TryChangeType(System.Object, System.Reflection.Binder, System.Globalization.CultureInfo, Boolean)
Stack trace:
at System.RuntimeType.TryChangeType(Object value, Binder binder, CultureInfo culture, Boolean needsSpecialCast)
at System.Reflection.MethodBase.CheckArguments(Object[] parameters, Binder binder, BindingFlags invokeAttr, CultureInfo culture, Signature sig)
at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, Object[] index)
at Rclr.ClrFacade.SetFieldOrProperty(Object obj, String name, Object value)
6.
clrCallStatic(clrFacadeTypeName, "SetFieldOrProperty", objOrType, 
name, value)
5.
rClr::clrSet(self$ref, propertyName, value) at dot-net-wrapper.R#47
4.
private$wrapProperty("ValuesAsArray", value) at data-column.R#15
3.
(function (value) 
{
private$wrapProperty("ValuesAsArray", value)
})(base::quote(0)) at data-set.R#297
2.
private$.setColumnValues(column = private$.xColumn, xValues) at data-set.R#243
1.
ds1$setValues(xValues = 0, yValues = 1, yErrorValues = 2.747)
```

This fails in core here:  

```
     /// <summary>
      ///    Required for R
      /// </summary>
      public virtual double[] ValuesAsArray
      {
         get => Values?.ToDoubleArray() ?? Array.Empty<double>();
         set => Values = value?.ToFloatArray();
      }
```

We need a function to set single values (how ugly, yes).

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Closes #889

### CODE REVIEW

File: R/sensitivity-analysis-results.R [hunk 1]
```diff
@@ -41,27 +42,72 @@ SensitivityAnalysisResults <- R6::R6Class("SensitivityAnalysisResults",
...
     #' @description
-    #' Returns the sensitivity value for a given pkParameter, output and parameter. If the sensitivity result does not exist, returns `NaN`
-    #' @param pkParameterName Name of pkParameter for which sensitivity should be retrieved
-    #' @param outputPath Path of the output for which the sensitivity should be retrieved
-    #' @param parameterName Name of the sensitivity parameter for which the sensitivity should be retrieved
-    pkParameterSensitivityValueFor = function(pkParameterName, outputPath, parameterName) {
-      value <- rClr::clrCall(self$ref, "PKParameterSensitivityValueFor", pkParameterName, outputPath, parameterName)
+    #'
+    #' Returns the sensitivity value for a given pkParameter, output and
+    #' parameter (either by path or by name). If the sensitivity result does not
```
Comment (PavelBal) [inline R/sensitivity-analysis-results.R:71]: "Parameter" lets call it "model parameter", to distinguish from "pk parameter".

**Review by PavelBal — CHANGES_REQUESTED** [review]
One small docs change and good to go.

**Comment by IndrajeetPatil** [comment]
@msevestre Tests still fail in `test-parameter.R` with the following error:

```csharp
Error (test-parameter.R:112:3): It can set a value in another unit and the value will be updated as expected
Error in `rClr::clrCallStatic(WITH_DIMENSION_EXTENSION, "ConvertToBaseUnit", 
    self$ref, value, unit)`: Type:    System.MissingMethodException
Message: Could not find a suitable static method ConvertToBaseUnit on type OSPSuite.Core.Domain.WithDimensionExtensions for method parameters DistributedParameter, Double, String
Method:  Void ThrowMissingMethod(System.Type, System.String, System.String, System.Type[])
Stack trace:
   at Rclr.ReflectionHelper.ThrowMissingMethod(Type classType, String methodName, String modifier, Type[] types)
   at Rclr.ClrFacade.InternalCallStaticMethod(Type classType, String methodName, Boolean tryUseConverter, Object[] arguments)
   at Rclr.ClrFacade.CallStaticMethod(String typename, String methodName, Object[] arguments)
```

Had a look at the corresponding [.NET method](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/develop/src/OSPSuite.Core/Domain/WithDimensionExtensions.cs), but couldn't figure out how to fix this.

**Comment by PavelBal** [comment]
@msevestre we need 

```
      public static **double** ConvertToBaseUnit(this IWithDimension withDimension, **double** value, string unit)
      {
         return HasUnit(withDimension, unit) ? ConvertToBaseUnit(withDimension, value, withDimension.Dimension.Unit(unit)) : value;
      }
```

it has been removed from `WithDimensionExtensions` for whatever reason and rClr cannot deal with float.

**Comment by msevestre** [comment]
@PavelBal ?
```
 public static double ConvertToUnit(this IWithDimension withDimension, double valueInBaseUnit, string unit)
      {
         return HasUnit(withDimension, unit) ? ConvertToUnit(withDimension, valueInBaseUnit, withDimension.Dimension.Unit(unit)) : valueInBaseUnit;
      }
```
It's here

**Comment by msevestre** [comment]
or here
```
  public static double ConvertToBaseUnit(this IWithDimension withDimension, double value, Unit unit)
      {
         return withDimension.Dimension?.UnitValueToBaseUnitValue(unit, value) ?? value;
      }

```

**Comment by msevestre** [comment]
oh with unit as string. Weird..

File: packages.centOS7.config [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="OSPSuite.FuncParser.CentOS7" version="4.0.0.50"/>
-  <package id="OSPSuite.SimModel.CentOS7" version="4.0.0.45"/>
+  <package id="OSPSuite.FuncParser.CentOS7" version="4.0.0.54"/>
```
Comment (msevestre) [inline packages.centOS7.config:3]: @Yuri05 Here

File: packages.config [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="OSPSuite.Assets" version="11.0.153"/>
-  <package id="OSPSuite.Core" version="11.0.153"/>
-  <package id="OSPSuite.Infrastructure" version="11.0.153"/>
-  <package id="OSPSuite.Infrastructure.Import" version="11.0.153"/>
-  <package id="OSPSuite.Infrastructure.Autofac" version="11.0.153"/>
-  <package id="OSPSuite.R" version="11.0.153"/>
+  <package id="OSPSuite.Assets" version="11.0.178"/>
```
Comment (msevestre) [inline packages.config:3]: @Yuri05 Here

File: packages.ubuntu18.config [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="OSPSuite.FuncParser.Ubuntu18" version="4.0.0.50"/>
-  <package id="OSPSuite.SimModel.Ubuntu18" version="4.0.0.45"/>
+  <package id="OSPSuite.FuncParser.Ubuntu18" version="4.0.0.54"/>
+  <package id="OSPSuite.SimModel.Ubuntu18" version="4.0.0.50"/>
```
Comment (msevestre) [inline packages.ubuntu18.config:4]: @Yuri05 Here

File: packages.win.config [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="OSPSuite.FuncParser" version="4.0.0.50"/>
-  <package id="OSPSuite.SimModel" version="4.0.0.45"/>
+  <package id="OSPSuite.FuncParser" version="4.0.0.54"/>
```
Comment (msevestre) [inline packages.win.config:3]: @Yuri05 Here

**Comment by msevestre** [comment]
Once again a pandoc error. Merging

**Pull request merged**